localstack/services/cloudwatch/cloudwatch_database_helper.py

```diff
@@ -0,0 +1,405 @@
+            if namespace:
+                namespace_filter = "AND namespace = ?"
+                data = data + (namespace,)
+
+            metric_name_filter = ""
+            if metric_name:
+                metric_name_filter = "AND metric_name = ?"
+                data = data + (metric_name,)
+
+            dimension_filter = ""
+            for dimension in dimensions:
+                dimension_filter += "AND dimensions LIKE ? "
+                data = data + (f"%{dimension.get('Name')}={dimension.get('Value','')}%",)
+
+            query = f"""
+                SELECT DISTINCT metric_name, namespace, dimensions
+                FROM (
+                    SELECT metric_name, namespace, dimensions, account_id, region, timestamp
+                    FROM SINGLE_METRICS
+                    UNION
+                    SELECT metric_name, namespace, dimensions, account_id, region, timestamp
+                    FROM AGGREGATED_METRICS
+                ) AS combined
+                WHERE account_id = ? AND region = ?
+                {namespace_filter}
+                {metric_name_filter}
+                {dimension_filter}
+                ORDER BY timestamp DESC
+            """
+
+            cur.execute(
+                query,
+                data,
+            )
+            metrics_result = [
+                {
+                    "metric_name": r[0],
+                    "namespace": r[1],
+                    "dimensions": self._restore_dimensions_from_string(r[2]),
+                }
+                for r in cur.fetchall()
+            ]
+
+            cur.execute(
+                query,
+                data,
+            )
+
+            return {"metrics": metrics_result}
+
+    def clear_tables(self):
+        # TODO clear tables for reset calls on cloudwatch
+        pass
+
+    def shutdown(self):
+        # TODO delete tmpdir/database if we do not have persistence enabled?
+        # anything else we should consider?
+        ...
+
+    def _get_ordered_dimensions_with_separator(self, dims: Optional[List[Dict]]):
+        if not dims:
+            return None
+        dims.sort(key=lambda d: d["Name"])
+        dimensions = ""
+        for d in dims:
+            dimensions += f"{d['Name']}={d['Value']}\t"  # aws does not allow ascii control characters, we can use it a sa separator
+
+        return dimensions
+
+    def _restore_dimensions_from_string(self, dimensions: str):
+        if not dimensions:
+            return None
+        dims = []
+        for d in dimensions.split("\t"):
+            if not d:
+                continue
+            name, value = d.split("=")
+            dims.append({"Name": name, "Value": value})
+
+        return dims
+
+    def _convert_timestamp_to_unix(
+        self, timestamp: datetime
+    ):  # TODO verify if this is the standard format, might need to convert
+        return int(timestamp.timestamp())
+
+    def _get_insert_single_metric_query(self) -> str:
```

Comment:
TODO: remove this, and use the insert-statement directly; same for `_get_insert_aggregated_metric_query`